sql: flake in TestAuthenticateWithSessionRevivalToken

Found in CI on an unrelated PR here: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/5821038?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildDeploymentsSection=true&expandBuildTestsSection=true

**Note: I can reproduce this reliably via a stress test on the `master` branch.**
(Took 500 iterations on first try)

```
Failed
=== RUN   TestAuthenticateWithSessionRevivalToken/authenticate_with_token
    session_revival_test.go:80:
          Error Trace:  session_revival_test.go:80
          Error:        Received unexpected error:
                        failed to connect to `host=127.0.0.1 user=testuser database=`: server error (ERROR: invalid session revival token: timestamp: nil Timestamp (SQLSTATE 28000))
          Test:         TestAuthenticateWithSessionRevivalToken/authenticate_with_token
    --- FAIL: TestAuthenticateWithSessionRevivalToken/authenticate_with_token (0.02s)

Failed
=== RUN   TestAuthenticateWithSessionRevivalToken/generate_token
    session_revival_test.go:64:
          Error Trace:  session_revival_test.go:64
          Error:        Received unexpected error:
                        ERROR: crdb_internal.create_session_revival_token(): session revival tokens are not supported on this cluster (SQLSTATE 0A000)
          Test:         TestAuthenticateWithSessionRevivalToken/generate_token
    --- FAIL: TestAuthenticateWithSessionRevivalToken/generate_token (0.05s)
```

cc @rafiss for triage

Jira issue: CRDB-17936

--- a/pkg/ccl/testccl/sqlccl/session_revival_test.go
+++ b/pkg/ccl/testccl/sqlccl/session_revival_test.go
@@ -17,6 +17,7 @@ import (
 	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/sessiondatapb"
 	"github.com/cockroachdb/cockroach/pkg/sql/tests"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
@@ -43,8 +44,7 @@ func TestAuthenticateWithSessionRevivalToken(t *testing.T) {
 
 	_, err := tenantDB.Exec("CREATE USER testuser WITH PASSWORD 'hunter2'")
 	require.NoError(t, err)
-	_, err = mainDB.Exec(`ALTER TENANT ALL SET CLUSTER SETTING server.user_login.session_revival_token.enabled = true`)
-	require.NoError(t, err)
+	sql.AllowSessionRevival.Override(ctx, &tenant.ClusterSettings().SV, true)
 
 	var token string
 	t.Run("generate token", func(t *testing.T) {

